rs/boundary_node/rate_limits/canister/access_control.rs

```diff
@@ -50,3 +54,47 @@ impl<R: CanisterApi> ResolveAccessLevel for AccessLevelResolver<R> {
         AccessLevel::RestrictedRead
     }
 }
+
+pub struct WithAuthorization<T, R> {
+    inner: T,
+    access_resolver: R,
+}
+
+impl<T, R> WithAuthorization<T, R> {
+    pub fn new(inner: T, access_resolver: R) -> Self {
+        Self {
+            inner,
+            access_resolver,
+        }
+    }
+}
+
+impl<T: AddsConfig, R: ResolveAccessLevel> AddsConfig for WithAuthorization<T, R> {
+    fn add_config(
+        &self,
+        input_config: rate_limits_api::InputConfig,
+        time: Timestamp,
+    ) -> Result<(), AddConfigError> {
+        // Only privileged users can perform this operation
+        if self.access_resolver.get_access_level() != AccessLevel::FullAccess {
+            return Err(AddConfigError::Unauthorized);
+        }
+        // Perform the inner call only if authorized.
+        self.inner.add_config(input_config, time)
```

Comment:
default should be unauthorized
```suggestion
        if self.access_resolver.get_access_level() == AccessLevel::FullAccess {      
           // Perform the inner call only if authorized.
           return self.inner.add_config(input_config, time)
        }
       Err(AddConfigError::Unauthorized)
```